sysdeps/lemon/include/lemon/filesystem.h

```diff
@@ -0,0 +1,36 @@
+#ifndef FS_H
+#define FS_H
+
+#include <stddef.h>
+#include <sys/types.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C"{
+#endif
+
+#define FS_NODE_FILE 0x1
+#define FS_NODE_DIRECTORY 0x2
+#define FS_NODE_BLKDEVICE 0x3
+#define FS_NODE_SYMLINK 0x4
+#define FS_NODE_CHARDEVICE 0x5
+#define FS_NODE_MOUNTPOINT 0x8
+
+typedef struct lemon_dirent {
+	uint32_t inode; // Inode number
+	char name[128]; // Filename
+    uint32_t type;
+} lemon_dirent_t;
+
+int lemon_open(const char* filename, int flags);
+void lemon_close(int fd);
+int lemon_read(int fd, void* buffer, size_t count);
+int lemon_write(int fd, const void* buffer, size_t count);
+off_t lemon_seek(int fd, off_t offset, int whence);
+int lemon_readdir(int fd, uint64_t count, lemon_dirent_t* dirent);
```

Comment:
I think it would make sense to extract these functions into some `liblemon` (for example). You could link such a library into mlibc at compile time. This is what the `sigma` target also does. Let me know if you have any questions about such an approach.

Reply:
I moved a bunch of code to my liblemon library, however I kept those filesystem functions in the libc as they are used by `sys_open`, `sys_read`, etc.

Reply:
You probably want to inline the code from the `lemon_{read,write,open,close}` etc into lemon's `sys_xxx` functions directly, since you're just passing the arguments directly anyway.